packages/camera/android/src/main/java/io/flutter/plugins/camera/Camera.java

```diff
@@ -421,6 +421,8 @@ public void resumeVideoRecording(@NonNull final Result result) {
   }
 
   public void startPreview() throws CameraAccessException {
+    if (pictureImageReader == null || pictureImageReader.getSurface() == null) return;
```

Comment:
Drive-by code review....

Under what circumstances is it null?  It looks like it gets set in `open()` and null'd out in `close()`.  Is the error you're seeing that the surface is null, and if so, how/when does the surface get populated?

Reply:
I see this error: "android.view.Surface android.media.ImageReader.getSurface() on a null object reference" if the surface is not visible immediately after accepting the camera permissions, and the app crashes. If I reopen the app, I no longer see the error. I think that an extra check for null safety is not bad if it saves from crashes.

Reply:
@Gioxxy could you possibly update your pubspec for your app to point to a locally modified camera plugin with heavy instrumentation (Android logs) added, so you can look at the logs and see what order stuff is happening in?

```yaml
dependency_overrides:
  camera:
    git:
      url: git://github.com/Gioxxy/plugins.git
      ref: fix_android_null_surface
      path: packages/camera
```

I'd like to understand how it comes to be that `pictureImageReader` or `pictureImageReader.getSurface()` gets nulled out - maybe there's a better solution than a simple null check here.